chain/client/src/client_actor.rs

```diff
@@ -1412,6 +1412,20 @@ impl ClientActor {
             debug!(target: "client", tail_height = tail, "Dropping a block that is too far behind.");
             return;
         }
+        // drop the block if a) it is not requested, b) we already processed this height, c) it is not building on top of current head
```

Comment:
please also mention, that we have to do this check ASAP - as the non-requested block means that we can get spammed.